### PR TITLE
BDN9 keymap

### PR DESCRIPTION
**Pull request by lickel**
## Description

This is a BDN9 keymap that toggles between two core layers. It has VIA enabled by default.
It uses the RGB matrix to signify which layer is currently active.

It was originally cherry-picked from https://github.com/qmk/qmk_firmware/pull/15922

It depends upon https://github.com/qmk/qmk_firmware/pull/15923

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by zvecr** [comment]
Core changes need to be submitted as its own PR.

**Comment by lickel** [comment]
Yup. This is a draft, waiting for the core PR to merge

**Comment by lickel** [comment]
@zvecr the "awaiting_pr" tag is no longer accurate. I am not able to remove it myself.

File: keyboards/keebio/bdn9/keymaps/lickel/keymap.c [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        TG(_CUST2), _______,    _______,
+        _______,    _______,    _______,
+        _______,    _______,    _______
+    ),
+};
+
+// clang-format on
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+#ifdef RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/keebio/bdn9/keymaps/lickel/keymap.c:64]: Not a fan of using this, as been working to deprecate it, in general.

File: keyboards/keebio/bdn9/keymaps/lickel/keymap.c [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    }
+#endif
+    return state;
+}
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    uint8_t layer = get_highest_layer(layer_state);
+
+    switch (layer) {
+#ifdef RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/keebio/bdn9/keymaps/lickel/keymap.c:110]:
```suggestion
#if defined(RGBLIGHT_ENABLE) || defined(RGB_MATRIX_ENABLE)
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by lickel** [comment]
@drashna or @fauxpark anything else needed to merge this?

**Comment by drashna** [comment]
Thanks!

**Comment by lickel** [comment]
Thanks!

**Pull request merged**